src/WalkingModule/src/Module.cpp

```diff
@@ -1949,3 +1681,102 @@ bool WalkingModule::dcmSmoother(const iDynTree::Vector2 adaptedDCM,const iDynTre
+        m_isPitchActive=m_stepAdapter->isArmPitchActive();
+        if(!m_stepAdapter->UpdateDCMEstimator(m_stableDCMModel->getCoMPosition(),m_stableDCMModel->getCoMVelocity(),measuredZMP,m_comHeightTrajectory.front()))
+        {
+            yError() << "[WalkingModule::updateModule] Unable to to recieve DCM from pendulumEstimator";
+            return false;
+        }
+        m_dcmEstimatedI= m_stepAdapter->getEstimatedDCM();
+    }
+
+    if(m_useStepAdaptation)
+    {
+        runStepAdapterOutput output;
+        output.indexPush=m_indexPush;
+        output.zmpNominal=m_zmpNominal;
+        output.zmpAdjusted=m_zmpAdjusted;
+        output.isPushActive=m_isPushActive;
+        output.isRollActive=m_isRollActive;
+        output.kDCMSmoother=m_kDCMSmoother;
+        output.indexSmoother=m_indexSmoother;
+        output.impactTimeNominal=m_impactTimeNominal;
+        output.isPitchActive=m_isPitchActive;
+        output.kFootSmoother=m_kFootSmoother;
+        output.indexFootSmoother=m_indexFootSmoother;
+        output.impactTimeAdjusted=m_impactTimeAdjusted;
+        output.dcmPositionAdjusted=m_DCMPositionAdjusted;
+        output.dcmVelocityAdjusted=m_DCMVelocityAdjusted;
+        output.adaptedFootLeftTwist=m_adaptatedFootLeftTwist;
+        output.currentFootLeftTwist=m_currentFootLeftTwist;
+        output.pushRecoveryActiveIndex=m_pushRecoveryActiveIndex;
+        output.adaptedFootRightTwist=m_adaptatedFootRightTwist;
+        output.currentFootRightTwist=m_currentFootRightTwist;
+        output.adaptedFootLeftTransform=m_adaptatedFootLeftTransform;
+        output.currentFootLeftTransform=m_currentFootLeftTransform;
+        output.adaptedFootRightTransform=m_adaptatedFootRightTransform;
+        output.currentFootRightTransform=m_currentFootRightTransform;
+        output.adaptedFootLeftAcceleration=m_adaptatedFootLeftAcceleration;
+        output.currentFootLeftAcceleration=m_currentFootLeftAcceleration;
+        output.timeIndexAfterPushDetection=m_timeIndexAfterPushDetection;
+        output.adaptedFootRightAcceleration=m_adaptatedFootRightAcceleration;
+        output.currentFootRightAcceleration=m_currentFootRightAcceleration;
+        output.FootTimeIndexAfterPushDetection=m_FootTimeIndexAfterPushDetection;
+
+        runningStepAdapterInput input;
+        input.dT=m_dT;
+        input.time=m_time;
+        input.timeOffset=m_timeOffset;
+        input.leftStepList=m_jLeftstepList;
+        input.leftInContact=m_leftInContact;
+        input.leftFootprints=m_jleftFootprints;
+        input.rightStepList=m_jRightstepList;
+        input.rightInContact=m_rightInContact;
+        input.rightFootprints=m_jRightFootprints;
+        input.dcmSubTrajectories=m_DCMSubTrajectories;
+        input.dcmPositionSmoothed=m_DCMPositionSmoothed;
+
+        m_stepAdapter->runStepAdaptation(input,output,std::move(m_trajectoryGeneratorStepAdjustment),std::move(m_trajectoryGenerator));
```

Comment:
Here I've used the above mentioned method!

Reply:
If you do so, the ``WalkingModule`` cannot use the ``TrajectoryGenerator`` object anymore. See also https://github.com/MiladShafiee/walking-controllers/pull/4#discussion_r422980814